dune/grid/common/GridPartitioning.cpp

```diff
@@ -234,12 +234,78 @@ namespace Dune
 	ensureConnectedPartitions(grid, num_part, cell_part, recursive);
     }
 
-    void addOverlapLayer(const CpGrid& grid, const std::vector<int>& cell_part, 
-                         std::vector<std::set<int> >& cell_overlap, int mypart, bool all)
+/// \brief Adds cells to the overlap that just share a point with an owner cell.
```

Comment:
I am a little skeptical about adding corner-neighbours. I would assume that only face-neighbours are needed? On the other hand I guess it will not inflict any harm other than a small amount of extra computation, and it may make it easier to try other numerical schemes than FV later on. Are there Dune interface requirements that makes it necessary to include the corner-neighbours?